avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -156,6 +156,12 @@ param primaryAgentPoolProfile array
 @description('Optional. Define one or more secondary/additional agent pools.')
 param agentPools agentPoolType
 
+@description('Optional. Whether or not to use AKS Automatic mode.')
+param automatic bool = false
+
+@description('Optional. Maintenance window for the maintenance configuration.')
+param maintenanceWindow object = {}
```

Comment:
Instead of 2 parameters, it may make sense to convert this to a single `param maintenanceConfiguration object?` parameter (ideally using a UDT), and update the deployment below to have a condition like `module managedCluster_maintenanceConfigurations 'maintenance-configurations/main.bicep' = if (!empty(maintenanceConfiguration)) { (...) }`
This way you can dynamically update it later without the need of additional parameters. Also, if using a UDT you could then declare the `maintenanceWindow` parameter as mandatory IF the `maintainanceConfiguartion` would be defined.

Reply:
UDTs are required for params right?

Reply:
Required no, recommended yes, absolutely 🚀